### ISSUE TITLE
openerp report designer openoffice plugin error.

### ISSUE THREAD

**Issue by DeepSilver** [odoo/odoo#713]
openerp report designer openoffice plugin error while trying to open report from server..
***Impacted versions:***

 - 8.0 (master)

***Operating system:***

 - Windows x86

***Open Office version:***

 - X-ApacheOpenOffice_4.1.0_rev4

***Steps to reproduce:***

 1. Open Office Writer Menu - > OpenErp Report Designer
 2. From the menu select Modify Existing Report
 3. Choose any of the report available

***Current behavior:***

 -  Download Error.
The report could not be downloaded
URL seems to be unsupported one. 

***Expected behavior:***

 - Open the report to be edit

***Content of the error report logged in openerp_report_designer.log:***

2014-06-23 11:47:57,608 ERROR Traceback (most recent call last):
  File "C:\Downloads\X-ApacheOpenOffice_4.1.0_rev4\User\OpenOffice 4\user\uno_packages\cache\uno_packages\svbssqn.tmp_\openerp_report_designer.zip\package\OpenERPReport.py", line 2850, in btnOk_clicked
    oDoc2 = desktop.loadComponentFromURL(url, "openerp", 55, arr)
com.sun.star.lang.IllegalArgumentException: URL seems to be an unsupported one.

Screen cast of the current behavior
http://www.youtube.com/watch?v=yjlvB8Otm1s

**Comment by Panchai81** [comment]
I have the same problem. Any solutions yet?

**Comment by gdgellatly** [comment]
The fix is very simple, although time and space consuming.  You need to
rebuild libreoffice from source, building uno with 2.7 and you will want
30GB or more spare disk before starting. Best I've managed is 90 minute
build with latest processor and 32GB of RAM.

https://gist.github.com/hbrunn/6f4a007a6ff7f75c0f8b

Then you need to disable automatic updates for libreoffice.

**Comment by Panchai81** [comment]
Dear Mr. Gellatly,

 

Thank you for your advice.

I’ll take a look on it. But I’m afraid my skills are not enough to rebuild an entirely program.

Will there be a bug fix any time soon of the module?

 

I’m using Openoffice 4.1.1.1 now, would it be better to us Libreoffice? I haven’t tried yet.

 

Regards,

 

Jack

 

Van: Graeme Gellatly [mailto:notifications@github.com] 
Verzonden: zondag 1 februari 2015 23:46
Aan: odoo/odoo
CC: Panchai81
Onderwerp: Re: [odoo] openerp report designer openoffice plugin error. (#713)

 

The fix is very simple, although time and space consuming. You need to 
rebuild libreoffice from source, building uno with 2.7 and you will want 
30GB or more spare disk before starting. Best I've managed is 90 minute 
build with latest processor and 32GB of RAM. 

 <https://gist.github.com/hbrunn/6f4a007a6ff7f75c0f8b> https://gist.github.com/hbrunn/6f4a007a6ff7f75c0f8b 

Then you need to disable automatic updates for libreoffice.

**Comment by lukebranch** [comment]
@gdgellatly ,

Thanks for the information. Have you had success building from source using this gist? I had tried initially when @hbrunn first released this very useful script, but I was not able to get it working on my own clean instance of Ubuntu 14.04.

**Comment by gdgellatly** [comment]
Hi I was responding to a specific comment about libreoffice and ubuntu
14.04.  I know nothing about Openoffice except I could not make it work
reliably on Ubuntu 14.04 and had no need for it on 12.04.

**Comment by gdgellatly** [comment]
Certainly, at least 10 times on a number of different machines, clean
installs, my own messed up laptop, dirty old demo server, even a 13.10
instance.  Had a couple of failures, once disk space, once the echo script
left out some really needed whitespace or required manual editing of some
dependency for some reason.

Maybe I didn't follow it exactly, I tend to use small scripts like a recipe
rather than run directly, not really sure, but the only real thing I
remember having to do was then copying the 2 uno files to my virtual env /
packages / buildout eggs directory depending on the implementation.  Maybe
I manually edited the source files rather than use sed (if it uses sed),
but really I was surprised by just how easy it was.

**Comment by lukebranch** [comment]
@gdgellatly ,

Thanks for your advice. I'm going to give it another go on Saturday and work through it manually as you've described. I'll post back on the jamotion issue that refers to this issue once i've been able to replicate it on my own instance consistently to see if it can be closed.

**Comment by KNitinx** [comment]
Anybody got the solution for this issue . I have Same Problem . Using ubuntu 14.04 and openoffice

**Comment by ngthuan** [comment]
I have 2 case:
1. If modify an existing report, it occur error when report is downloaded to computer.
2. If I create a new report, then save on to server. Then I can modify it succesfully.

**Comment by lukebranch** [comment]
@KNitinx , @ngthuan ,

I suggest you try Aeroo Reports instead. Here's an install script for:

Odoo + Aeroo Reports

for Odoo 8:

https://github.com/OdooCommunityWidgets/install-scripts/blob/8.0/aeroo_reports/aeroo_install.sh

Aeroo reports can handle the same output as the original report designer did. I think the report designer is an unsupported module due to the fact that pyuno in Ubuntu 14.04 is written in Python 3+, while Odoo is written in Python 2.7+ making them incompatible.

There are many lengthy discussions on the subject that you can google (aeroo reports and python-uno) so I will not go into detail.

My suggestion is don't waste your time trying to get the report designer to work, and just use Aeroo Reports instead.

**Comment by ngthuan** [comment]
Hi,

Thank you for you knidly help.

My platform is windows 8. Open office 4.1.1

regards,

Thuan Nguyen

2015-11-18 16:19 GMT+07:00 Luke Branch <notifications@github.com>:

> @KNitinx <https://github.com/KNitinx> , @ngthuan
> <https://github.com/ngthuan> ,
>
> I suggest you try Aeroo Reports instead. Here's an install script for:
>
> Odoo + Aeroo Reports
>
> for Odoo 8:
>
>
> https://github.com/OdooCommunityWidgets/install-scripts/blob/8.0/aeroo_reports/aeroo_install.sh
>
> Aeroo reports can handle the same output as the original report designer
> did. I think the report designer is an unsupported module due to the fact
> that pyuno in Ubuntu 14.04 is written in Python 3+, while Odoo is written
> in Python 2.7+ making them incompatible.
>
> There are many lengthy discussions on the subject that you can google
> (aeroo reports and python-uno) so I will not go into detail.
>
> My suggestion is don't waste your time trying to get the report designer
> to work, and just use Aeroo Reports instead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/odoo/odoo/issues/713#issuecomment-157652627>.
>



-- 
Best regards,

Thuan Nguyen.
General manager
cell phone: 0915 859 856, email: ngthuan@khaian.com
==================================================
Khai An Co., Ltd.
Rosslare Security Distributor in Viet Nam
Tel/fax: 511 3643 644 / 3634 123
e/w: sales@khaian..com/ khaian.com

**Comment by lukebranch** [comment]
@ngthuan ,

Unfortunately I have zero experience with Odoo in Windows. My personal recommendation would be to use Ubuntu 14.04 for running Odoo + Aeroo in, as I think you will have a hell of a time trying to get Odoo 8.0 + Aeroo working running in a windows environment.

You can try to run it from an Ubuntu 14.04 (Server Edition) virtual machine inside windows with VirtualBox or similar if you want to try it out.

I can only offer you advice based on my experience with Odoo + Aeroo on Ubuntu 14.04 operating system.

**Comment by Yenthe666** [comment]
I agree with @lukebranch, you should really go for Ubuntu..

**Comment by Yenthe666** [comment]
Dear @DeepSilver,

Thank you for reporting this issue!
At this point Odoo 6.0, 6.1 and 7.0 are no longer supported for bugsfixes anymore, which means we will no longer handle these bugs.
If you still face this issue on Odoo 8, 9 or 10 we'd like to ask you to create a new issue on how to reproduce this.
Our apologies.

Regards,
Yenthe
